Update SDKs for google provider package

lwyszomi opened the pull request:
As everyone know google provider package have a lot of old dependencies. I would like to start migration to the latest versions of the SDK. For now we are blocked by some other dependencies because they are using `protobuf<4`.

```
apache-beam
mysql-connector-python
yandexcloud
```

Also in the google SDKs we had a lot of breaking changes so after updating we need to adjust broken operators. I did investigation how big is this problem and I'm attaching the list of services where some of the operators are broken:


- [x] AutoML -> need investigation
- [x] BigQuery -> need investigation
- [x] BigTable -> need investigation
- [x] CloudBuild -> need investigation
- [x] CloudFunctions -> need investigation
- [x] CloudMemorystore -> OK
- [x] CloudSQL -> need investigation
- [x] Composer -> adjust system tests
- [x] Compute -> need adjustments in system tests
- [x] DataLossPrevention -> OK
- [x] Dataflow -> need investigation
- [x] dataform -> OK
- [x] datafusion -> OK
- [x] dataplex -> need investigation
- [x] dataprep -> need investigation
- [x] Dataproc -> need investigation
- [x] dataprocMetastore -> need investigation
- [x] datastore -> need adjustments in system tests
- [x] GCS -> need investigations
- [x] kubernetesEngine -> OK
- [x] LifeSciences -> OK
- [x] MLEngine -> need investigation
- [x] NaturalLanguage -> need investigation
- [x] PubSub - OK
- [x] Spanner - OK
- [x] SpeachToTet -> need investigation
- [x] SQLToSheets -> need investigation
- [x] StackDriver -> need investigation 
- [x] StorageTransfer -> need investigation
- [x] Tasks -> need investigation
- [x] TextToSpeech -> need investigation
- [x] Transfers -> Need investigation
- [x] Translate -> OK
- [x] TranslateSpeech -> Need investigation
- [x] VertexAI - Need Investigation
- [x] VideoIntelligence - need investigation
- [x] Vision - need investigation
- [x] Workflows - OK


Fixes: #27292

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

Commit:
Update providers.yaml for google

kosteev committed:
Update BigTable operators to accomodate for new dependencies.

Commit:
Update SDK versions for Google provider

Commit:
Adjust google ads operators to v12

moiseenkov committed:
Fix BigQuery system tests

Changes:
- fix tests/system/providers/google/cloud/bigquery/example_bigquery_queries.py
- fix tests/system/providers/google/cloud/bigquery/example_bigquery_queries_async.py

VladaZakharova committed:
Update Compute Engine operators to accomodate for new dependencies.

Changes:
- added wait_for_operation_complete() method to check the execution flow
- added new attribute cmd_timeout for ComputeEngineSSHHook

Commit:
Update Translate Speech operators to accommodate new dependencies.

Changes:
- fix synthesize_speech method for CloudTextToSpeechHook
- fix CloudTranslateSpeechOperator
- tests/providers/google/cloud/operators/test_translate_speech.py

ahidalgob committed:
Fix Cloud Tasks System tests

Tasks dag was quite flaky without the retry option in the run_task step,
but it's consistently green with the option set.

We also add a GCP_APP_ENGINE_LOCATION env variable since this depends on
the used GCP Project App Engine's location

Commit:
Update SpeechToText operators to accommodate new dependencies.

Changes:
- fix synthesize_speech method for CloudTextToSpeechHook
- fix CloudSpeechToTextRecognizeSpeechOperator
- fix tests/providers/google/cloud/operators/test_speech_to_text.py
- fix tests/providers/google/cloud/hooks/test_text_to_speech.py
- fix tests/providers/google/cloud/hooks/test_speech_to_text.py

ahidalgob committed:
Add setup docstring to Tasks system tests.

Commit:
Update VideoIntelligence operators to accommodate new dependencies.

Changes:
- fix annotate_video method for CloudVideoIntelligenceHook
- fix VideoIntelligence Operators
- fix tests/providers/google/cloud/hooks/test_video_intelligence.py
- fix tests/providers/google/cloud/operators/test_video_intelligence.py

Commit:
Update Vision operators to accommodate new dependencies.

Changes:
- fix methods for CloudVisionHook
- fix Vision Operators
- fix tests/providers/google/cloud/hooks/test_vision.py
- fix tests/providers/google/cloud/operators/test_vision.py
- fix tests/system/providers/google/cloud/vision/example_vision_annotate_image.py
- fix tests/system/providers/google/cloud/vision/example_vision_autogenerated.py
- fix tests/system/providers/google/cloud/vision/example_vision_explicit.py

Commit:
Update Natural Language operators to accommodate new dependencies.

Changes:
- fix airflow/providers/google/cloud/operators/natural_language.py
- fix airflow/providers/google/cloud/hooks/natural_language.py
- fix tests/providers/google/cloud/hooks/test_natural_language.py
- fix tests/providers/google/cloud/operators/test_natural_language.py
- fix tests/system/providers/google/cloud/natural_language/example_natural_language.py

VladaZakharova committed:
Update AutoML operators to accommodate new dependencies.

Changes:
- add timeout parameter to all long-running operations for operators
- fix tests/system/providers/google/cloud/automl/example_automl_dataset.py
- fix tests/system/providers/google/cloud/automl/example_automl_model.py
- fix tests/system/providers/google/cloud/automl/example_automl_nl_text_extraction.py
- fix tests/system/providers/google/cloud/automl/example_automl_vision_classification.py

ahidalgob committed:
Fix Stackdriver system test

This test has not worked because of slack channel and credentials not
being setup. We now test the same operators by creating notification
channels and policy alerts against pubsub topics, which don't need to
exist before the test is ran, making the test self-contained.

kosteev committed:
Update Composer system tests.

Fix environment id to contain underscores.

VladaZakharova committed:
Update VertexAI operators to accommodate new dependencies.

ahidalgob committed:
Fix Cloud SQL delete operator

For some delete instance operations, the operation stops being available ~9 seconds after completion, so we need a shorter sleep time to make sure we don'tmiss the DONE status.

moiseenkov committed:
Fix GCS system tests

ahidalgob committed:
Add SQL to Sheets Test instructions

moiseenkov committed:
Fix CloudBuild unit test

Commit:
Update Dataproc Metastore operators to accommodate new dependencies.

Commit:
Update Dataproc operators to accommodate new dependencies.

ahidalgob committed:
Update Dataflow sys tests to new sdk

Commit:
Update Dataproc on gke operators to accommodate new dependencies.

VladaZakharova committed:
Update MLEngine operators to accomodate new dependencies.

Changes:
- update train model that is used for prediction
- update version and runner for ApacheBeam in utils for MLEngine
- update connection inside async hook

Commit:
Update Dataprep operators to accommodate new dependencies.

Changes:
- fix tests/system/providers/google/cloud/dataprep/example_dataprep.py

ahidalgob committed:
Add Dataflow Go system test

potiuk committed:
fixup! Update providers.yaml for google

r-richmond commented on airflow/providers/apache/beam/provider.yaml:
```diff
@@ -42,7 +42,7 @@ versions:
 dependencies:
   - apache-airflow>=2.3.0
-  - apache-beam>=2.33.0
+  - apache-beam==2.47.0rc1
```
```suggestion
  - apache-beam>=2.47.0rc1
```
@potiuk Should it be this instead?

potiuk replied:
Finally it should be likely >= 2.47 when they release. For now - I wanted to make sure it works, and put ==  to unblock @lwyszomi and the team to start solving test failures as the image could not be built without it due to conflicts.

And we are in a good place finally. The image builds without conflicts :tada: !!! and those are the upgraded packages  -- exactly what we wanted :) 

```
< # This constraints file was automatically generated on 2023-04-27T21:53:27Z
---
> # This constraints file was automatically generated on 2023-04-27T23:10:05Z
43a44
> Shapely==1.8.5.post1
63c64
< apache-beam==2.46.0
---
> apache-beam==2.47.0rc1
108c109
< black==23.1a1
---
> black==23.3.0
162c163
< elasticsearch-dbapi==0.2.10
---
> elasticsearch-dbapi==0.2.2
164c165
< elasticsearch==7.13.4
---
> elasticsearch==7.17.9
189,190c190,192
< google-api-core==2.8.2
< google-api-python-client==1.12.11
---
> google-ads==21.0.0
> google-api-core==2.11.0
> google-api-python-client==2.86.0
192c194
< google-auth-oauthlib==0.8.0
---
> google-auth-oauthlib==1.0.0
194,204c196,206
< google-cloud-aiplatform==1.16.1
< google-cloud-appengine-logging==1.1.3
< google-cloud-audit-log==0.2.4
< google-cloud-automl==2.8.0
< google-cloud-bigquery-datatransfer==3.7.0
< google-cloud-bigquery-storage==2.14.1
< google-cloud-bigquery==2.34.4
< google-cloud-bigtable==2.11.1
< google-cloud-build==3.9.0
< google-cloud-compute==0.7.0
< google-cloud-container==2.11.1
---
> google-cloud-aiplatform==1.24.1
> google-cloud-appengine-logging==1.3.0
> google-cloud-audit-log==0.2.5
> google-cloud-automl==2.11.1
> google-cloud-bigquery-datatransfer==3.11.1
> google-cloud-bigquery-storage==2.19.1
> google-cloud-bigquery==3.10.0
> google-cloud-bigtable==2.17.0
> google-cloud-build==3.16.0
> google-cloud-compute==1.11.0
> google-cloud-container==2.21.0
206,225c208,227
< google-cloud-datacatalog==3.9.0
< google-cloud-dataflow-client==0.5.4
< google-cloud-dataform==0.2.0
< google-cloud-dataplex==1.1.0
< google-cloud-dataproc-metastore==1.6.0
< google-cloud-dataproc==5.0.0
< google-cloud-dlp==3.8.0
< google-cloud-kms==2.12.0
< google-cloud-language==1.3.2
< google-cloud-logging==3.2.1
< google-cloud-memcache==1.4.1
< google-cloud-monitoring==2.11.0
< google-cloud-orchestration-airflow==1.4.1
< google-cloud-os-login==2.7.1
< google-cloud-pubsub==2.13.5
< google-cloud-redis==2.9.0
< google-cloud-resource-manager==1.6.0
< google-cloud-secret-manager==1.0.2
< google-cloud-spanner==1.19.3
< google-cloud-speech==1.3.4
---
> google-cloud-datacatalog==3.12.0
> google-cloud-dataflow-client==0.8.3
> google-cloud-dataform==0.5.1
> google-cloud-dataplex==1.4.3
> google-cloud-dataproc-metastore==1.11.0
> google-cloud-dataproc==5.4.1
> google-cloud-dlp==3.12.1
> google-cloud-kms==2.16.1
> google-cloud-language==2.9.1
> google-cloud-logging==3.5.0
> google-cloud-memcache==1.7.1
> google-cloud-monitoring==2.14.2
> google-cloud-orchestration-airflow==1.7.1
> google-cloud-os-login==2.9.1
> google-cloud-pubsub==2.16.0
> google-cloud-redis==2.12.1
> google-cloud-resource-manager==1.10.0
> google-cloud-secret-manager==2.16.1
> google-cloud-spanner==3.32.0
> google-cloud-speech==2.19.0
227,232c229,234
< google-cloud-tasks==2.10.1
< google-cloud-texttospeech==1.0.3
< google-cloud-translate==1.7.2
< google-cloud-videointelligence==1.16.3
< google-cloud-vision==1.0.2
< google-cloud-workflows==1.7.1
---
> google-cloud-tasks==2.13.1
> google-cloud-texttospeech==2.14.1
> google-cloud-translate==3.11.1
> google-cloud-videointelligence==2.11.1
> google-cloud-vision==3.4.1
> google-cloud-workflows==1.10.1
235c237
< googleapis-common-protos==1.56.4
---
> googleapis-common-protos==1.59.0
239c241
< grpc-google-iam-v1==0.12.4
---
> grpc-google-iam-v1==0.12.6
241c243
< grpcio-status==1.48.2
---
> grpcio-status==1.54.0
356,357c358,359
< packaging==21.3
< pandas-gbq==0.17.9
---
> packaging==23.1
> pandas-gbq==0.19.1
385,386c387,388
< proto-plus==1.19.6
< protobuf==3.20.0
---
> proto-plus==1.22.2
> protobuf==4.22.3
394c396
< pyarrow==9.0.0
---
> pyarrow==11.0.0
512c514
< tableauserverclient==0.24
---
> tableauserverclient==0.25
558c560
< uritemplate==3.0.1
---
> uritemplate==4.1.1
```

ahidalgob committed:
Google SDK Fixes after rebase

- Secret Manager was missing updating to v2, now expects a request dict
- Compute ssh had a bug when no cmd_timeout was passed
- Cloud Build tests were improved/refactored in community, so deleting
old ones
- googleapiclient.errors.HttpError was incorrectly used in our tests, it
it didn´t matter before but a change in the class makes HttpError()
raise an error in initialization the way we were using it before
- fix static checks

```
$ pytest tests/providers/google/cloud/
...
===== 2763 passed, 71 skipped, 21 warnings in 193.46s (0:03:13) =====
```

ahidalgob committed:
Fix Google providers type errors

kosteev commented on airflow/providers/google/provider.yaml:
```diff
@@ -67,74 +67,59 @@ versions:
   - sqlalchemy-bigquery>=1.2.1
+  # A transient dependency of google-cloud-bigquery-datatransfer, but we
+  # further constrain it since older versions are buggy.
+  - proto-plus>=1.19.6
```
Can you, please, put it in lexicographical order
